src/main/kotlin/no/skatteetaten/aurora/fergus/service/StorageGridService.kt

```diff
@@ -26,11 +59,293 @@ class StorageGridServiceReactive(private val storageGridAuthApi: AuthApi) : Stor
         }
         return response.data // Returns authorization token
     }
+
+    override suspend fun provideBucket(
+        bucketName: String,
+        token: String
+    ): String {
+        storageGridContainersApi.apiClient.setBearerToken(token)
```

Comment:
Med singleton scope vil endring av token gjelde andre pågående requests også?

Reply:
Vi bør vurdere request-scope hvis det er slik at tokenet skal oppdateres for hver forespørsel fra klient?

Reply:
Jeg vil tro at det kanskje er slik at de ulike StorageGrid-apiene kanskje bruker samme apiClient, men jeg har valgt å ta med token eksplisitt for alle kall til service der det trengs. Dette med tokens må nok gås opp på nytt når man skal opprette caching for tokens og kanskje også S3-credentials.

Det er for øvrig en viktig sak å sikre at man får trådsikkerhet her. Siden vi har en viss asynkronitet, så må vi sikre at token for en bruker ikke overskrives i apiClient om det kommer kall fra en annen bruker.

Reply:
Fikset.